### ISSUE TITLE
[Bug]: Spinner focusing and screen readers issues

### ISSUE THREAD

**Issue by kolaps33** [microsoft/fluentui#31025]
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
System:
    OS: Windows 10 10.0.22631
    CPU: (12) x64 Intel(R) Xeon(R) E-2176M  CPU @ 2.70GHz
    Memory: 17.28 GB / 31.74 GB
  Browsers:
    Edge: Chromium (123.0.2420.81)
    Internet Explorer: 11.0.22621.1
```

### Are you reporting Accessibility issue?

None

### Reproduction

https://stackblitz.com/edit/68jtch?file=src%2Fexample.tsx

### Bug Description

Run NVDA or JAWS 
Navigate to sample code:
1. https://stackblitz.com/edit/68jtch?file=src%2Fexample.tsx
2. set focus on test button 
3. Press TAB key navigate to the spinner 

## Actual Behavior
screen reader is silent

## Expected Behavior
screen reader should announce the progressbar element with corresponding aria-label

issue is that role and aria-label is on the different element like the tabindex itself: 
![image](https://github.com/microsoft/fluentui/assets/22620150/cedd15ec-3b26-4a3c-b3dd-33dcdf90f3a1)
All attributes should be on the same element

### Requested priority

High

### Are you willing to submit a PR to fix?

yes

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by kolaps33** [comment]
As well 
when the spinner has tabindex set on wrong element
when user get `ref` into the spinner,  
then the spinner is not focusable as `<div>` has no tabindex element.

Please see example:
https://stackblitz.com/edit/omex1r-3qxggh?file=src%2Fexample.tsx

in example there is done hack in order to set focus on spinner

**Comment by micahgodbolt** [comment]
I will have Tomi look what is the correct application for aria-label and tabIndex. I agree they should both go on the same element, but it seems tabIndex was specifically placed on the spinner slot.

For now, one option would be to apply both tabIndex and aria-label to the spinner via:

```jsx
    <Spinner spinner={{tabIndex: 0, 'aria-label': 'loading...'}}  {...props} />
```

**Comment by smhigley** [comment]
Since the root slot is also the one that gets a default `aria-labelledby` and is the primary slot, I think it makes the most sense  for `tabindex` to go there. The current behavior does look intentional enough that Tomi should weigh in first though.

**Comment by tomi-msft** [comment]
The `tabindex` should actually be in the root slot, I will make a PR to update this.
